fecgov/fec-cms
# [WIP] Update removeAllTags to have default values

## Summary

- Resolves #2987 

## Impacted areas of the application
Added some default values to removeAllTags inside filter-tags.js and triggered fake events on those elements to more reliably integrate with the current structure.

## Screenshots
None

## Related PRs
None

## How to test
- Pull the branch
- `npm run build`
- `npm run test-single`
- Make sure no tests failed
- Check the receipts, individual contributions, and other datatables'
  - 'clear all filters' link text (should read "Reset filters" for Receipts and Individual Contributions but remain "Clear all filters" for every other use)
  - default states (data present, no errors, no stalls)
  - state as data is filtered down (use some non-date filters) (data and filters update appropriately)
  - state when the date filters are removed from the top of the tables
    - after the filters are cleared [1]
  - state when the date filters are deselected/removed from the left/filters
    - after the filters are cleared [1]
  - state when the Report Time Period tag/label is totally removed from the filters
    - after the filters are cleared [1]
- *Anywhere else "Clear all filters" is used that we may have missed?*

[1]: Receipts and Individual contributions tables should revert to their default state, i.e. with 2019-2020 and 01/01/2019-12/31/2020 filters applied. Other tables should clear all filters.

# Question / Known issue
[Receipts](http://127.0.0.1:8000/data/receipts/?data_type=processed&two_year_transaction_period=2020&min_date=01%2F01%2F2019&max_date=12%2F31%2F2020) and [disbursements](http://127.0.0.1:8000/data/disbursements/?data_type=processed&two_year_transaction_period=2020&min_date=01%2F01%2F2019&max_date=12%2F31%2F2020) default with an "All report years" filter applied, but clicking Clear all filters now changes it to "2019-2020" (calculated from DEFAULT_ELECTION_YEAR)

Is this okay? Do we need a more solid option?


____
